Automattic/wp-calypso
# Site Settings: Remove discussion section wrapper

This cleanup PR updates the Discussion Settings form to not use the legacy `site` object, but rather to rely on the selected site ID, which comes from Redux. This way the `SiteSettingsDiscussion` section component becomes totally obsolete, so it can be removed and `form-discussion` can be loaded directly instead. Also, `wrapSettingsForm` should use connected prop `siteId` instead of assuming there will be a `site.ID` passed to the target component.

To test:

* Checkout this branch.
* Head to `/settings/discussion/$site` for a WP.com site.
* Verify settings are loaded and saved as expected.
* Verify the above for a Jetpack site.
